# <center>Практическое задание к уроку №1

**1) Из колоды в 52 карты извлекаются случайным образом 4 карты.**   
       
      a) Найти вероятность того, что все карты – крести. 
      б) Найти вероятность, что среди 4-х карт окажется хотя бы один туз. 
   
---

a) В колоде из 52 карт $\frac{52}{4}=13$ карт одной масти.  
Число сочетаний из $n$ элементов по $k$ элементов в каждом:  
$C^k_n=\frac{n!}{k!(n-k)!}$  

Применяем формулу вероятности:  

$P(A)=\frac{m}{n}$ где $m$-число **благоприятных** исходов, $n$-общее число **всех** исходов.  
в нашем случае $n=C^4_{52}$-т.е. количество способов извлечь 4 карты из 52, а $m=C^4_{13}$, способов извлечь 4 карты одной масти.  
Создадим функцию:

In [6]:
from math import factorial
def combinations(n, k):
    return int(factorial(n) / (factorial(k) * factorial(n - k))) 

Вычисляем $m$:

In [7]:
m = combinations(13, 4)
print(f'm = {m}')

m = 715


Вычисляем $n$:

In [8]:
n = combinations(52, 4)
print(f'n = {n}')

n = 270725


Находим вероятность $P(A)$, где $А$ вероятность того что все 4-ре карты - крести:

In [11]:
P = m/n
print(f'P(A) = {round(P, 4)}')

P(A) = 0.0026


Из формулы вероятности следует, что вероятность извлечь карту из 4-х:

1-ю крести: $\displaystyle\frac{13}{52}$;  

2-ю крести: $\displaystyle\frac{12}{51}$;  

3-ю крести: $\displaystyle\frac{11}{50}$;  

4-ю крести: $\displaystyle\frac{10}{49}$.  

Так как, по условию, все карты должны быть крести, то $\displaystyle P(A)=\frac{13}{52}\cdot\frac{12}{51}\cdot\frac{11}{50}\cdot\frac{10}{49}$  

Вычислим:     

In [12]:
P1 = 13/52 * 12/51 * 11/50 * 10/49
print(f'P(A) = {round(P1,4)}')

P(A) = 0.0026


----
б) Найти что среди карт окажется хотя бы один туз: $\displaystyle n=C^4_{52}$  
где $\displaystyle m$ будет складываться из 4 вариантов:  
   
1. Из 4-х карт 1 туз:  
   число способов достать 1 туз из 4-х $C^1_4$, число способов достать три остальные карты $С^3_{48}$, общее количество сочетаний 1 туз и 3 не туза $C^1_{4}\cdotС^3_{48}$


2. Из 4-х карт 2 туза:  
   число способов достать 2 туза из 4-х $C^2_4$, число способов достать две остальные карты $С^2_{48}$, общее количество сочетаний 2 туза и 2 не туза $C^2_{4}\cdotС^2_{48}$


3. Из 4-х карт 3 туза:  
   число способов достать 3 туза из 4-х $C^3_4$, число способов достать оставшуюся карту $С^1_{48}$, общее количество сочетаний 3 туза и 1 не туз $C^3_{4}\cdotС^1_{48}$


4. Из 4-х карт 4 туза:  
   число способов достать 4 туза из 4-х $C^4_4=1$  
   

Общее число благоприятных исходов - сумма этих 4 вариантов.  

Вычислим:  

In [14]:
n = combinations(52, 4)
print(f'n = {n}')

n = 270725


In [15]:
m = sum([combinations(4,1)*combinations(48,3),combinations(4,2)*combinations(48,2),combinations(4,3)*combinations(48,1),1])
print(f'm = {m}')

m = 76145


Найдем вероятность $P(A)$, где $A$ - вероятность достать хотя бы одного туза:

In [16]:
P2=1-combinations(48,4)/combinations(52,4)
print(f'P(A) = {round(P2,4)}')

P(A) = 0.2813


#### Ответ:  
а) $P(\text{A1})\approx 0.0026 \approx 0.26\%$;  
б) $P(\text{A2})\approx 0.2813 \approx 28.13\%$.  

**2) На входной двери подъезда установлен кодовый замок, содержащий десять кнопок с цифрами от 0 до 9. Код содержит три цифры, которые нужно нажать одновременно. Какова вероятность того, что человек, не знающий код, откроет дверь с первой попытки?**  

---

Так как кнопки нажимаются одновременно, то порядок не важен, и для формулы вероятности $\displaystyle P(A)=\frac{m}{n}$ общее число исходов будет определяться количиством сочетаний 3-х элементов из 10, $\displaystyle n=C^3_{10}$, а количество благоприятных исходов $\displaystyle m=1$.

Вычислим $\displaystyle P(A)$:  

In [19]:
P = 1/combinations(10, 3)
print(f'P(A) = {round(P, 4)}')

P(A) = 0.0083


___

#### Ответ:

$\displaystyle P(A)\approx0.0083\approx0.83\%$  

___

**3) В ящике имеется 15 деталей, из которых 9 окрашены. Рабочий случайным образом извлекает 3 детали. Какова вероятность того, что все извлеченные детали окрашены?**  

---

Общее количество исходов будет определяться количеством сочетаний 3 деталей из 15: $\displaystyle n=C^3_{15}$, а количество благоприятных исходов - количеством сочетаний 3-х окрашенных деталей из 9:  $\displaystyle m=C^3_{9}$  

Вычислим $P(A)$, где $А$ - вероятность что все детали окрашены:

In [20]:
P = combinations(9, 3)/combinations(15, 3)
print(f'P(A) = {round(P, 4)}')

P(A) = 0.1846


---

#### Ответ:  
$P(A)\approx0.1846\approx18.46\%$  

---

**4) В лотерее 100 билетов. Из них 2 выигрышных. Какова вероятность того, что 2 приобретенных билета окажутся выигрышными?**  

---

Найдем вероятность $P(A)$, где $A$ - вероятность, что 2 билета выигрышные: 

$\displaystyle P(A)=\frac{1}{C^2_{100}}=\frac{1}{\frac{100!}{2!(100-2)!}}=\frac{2!\cdot98!}{100!}$ 

Вычислим:

In [23]:
P = 1/combinations(100, 2)
print(f'P(A) = {round(P, 4)}')

P(A) = 0.0002


___

#### Ответ:

$P(A)\approx0.0002\approx0.2\%$

___